source/core_http_client.c

```diff
@@ -2237,7 +2277,7 @@ static int findHeaderFieldParserCallback( http_parser * pHttpParser,
     /* Check whether the parsed header matches the header we are looking for. */
     /* Each header field consists of a case-insensitive field name (RFC 7230, section 3.2). */
     if( ( fieldLen == pContext->fieldLen ) &&
-        ( strncasecmp( pContext->pField, pFieldLoc, fieldLen ) == 0 ) )
+        ( caseInsensitiveStringCmp( pContext->pField, pFieldLoc, fieldLen ) == 0 ) )
```

Comment:
Rather than create a replacement for `strncasecmp()`, why not just convert both strings to lower-case first and then use `strncmp()`?

Reply:
As suggested by you in the comment ,To convert strings to lowercase first and then compare using strncmp, we have to either use strlwr function which is again a non-standard C function and may not be compatible across different platforms or we have to do it character by character using tolower function so for that also we need to have a separate function to loop through the entire string so thats why I thought of having a custom implemented caseInsesitiveStringCompare function which will do both casechange and comparison. Also converting to lowercase first and then comparing will modify the original string which is not required. Let me know if its good to go with this approach.